blog/2024-10-29-vibration-analysis-python/index.mdx

```diff
@@ -0,0 +1,376 @@
+
+$$\text{RMS} = \sqrt{\frac{1}{n} \sum_{i=1}^{n} x_i^2}$$
+
+The underlying concept of energy is the reason why the formula includes an inner square (inner product of the vector of itself) and square root: since averaging the raw amplitude is meaningless (and leads to near zero values for balanced signal), we use the equivalent constant amplitude that matches the original signal's average energy. For instance, in an electric signal, RMS is just the value of direct current that would dissipate the same amount of heat (have the same power, which is a sum of per-point energies divided by the window length) as the original signal over the time window.
+
+As we’ll see later, RMS can also be derived from the frequency domain by taking the square root of the area under the curve of spectral power density.
+
+In Python, we can calculate RMS as follows:
+
+```python
+import numpy as np
+import numpy as np
+import matplotlib.pyplot as plt
+
+np.random.seed(0)
+
+signal = 0.5 * np.random.randn(1000)  # standard deviation 0.5
+
+rms = np.sqrt(np.mean(signal**2))
+
+plt.plot(signal, c="indigo")
+plt.axhline(y=rms, color='red', linestyle='--', label=f"RMS: {rms:.2f}")
+plt.title(f"Signal with RMS: {rms:.2f}")
+plt.xlabel("Sample")
+plt.ylabel("Amplitude")
+plt.show()
+```
+
+![RMS plot of zero-centered Gaussian noise, stddev=0.5](./img/rms_noise.png)
+
+As we could expect for the normally distributed zero-centered noise with a standard deviation of 0.5, we get an RMS value also about 0.5.
+
+If we add to the signal any summand, we'll see in RMS its additional energy:
+
+```python
+import numpy as np
+import matplotlib.pyplot as plt
+
+np.random.seed(0)
+
+noise = 0.5 * np.random.randn(1000)  # the same as above, sigma = 0.5
+spike = np.zeros(1000)
+spike[500:520] = 5  # add DC offset 5 for 20 points
+signal = noise + spike
+
+rms = np.sqrt(np.mean(signal**2))
+
+plt.plot(signal, c="indigo")
+plt.axhline(y=rms, color='red', linestyle='--', label=f"RMS: {rms:.2f}")
+plt.title(f"Signal with RMS: {rms:.2f}")
+plt.xlabel("Sample")
+plt.ylabel("Amplitude")
+plt.show()
+```
+
+![RMS plot with peak](./img/rms.png)
+
+### Peak amplitude and crest factor
+
+Windowed **peak amplitude** is simply the highest absolute amplitude in the window, and it tells nothing about all other points of the signal than the highest peak. A more informative feature is the ratio of peak energy to average energy, i.e. the squared peak amplitude divided by the squared RMS, known as the **crest factor**. The crest factor indicates how much the peak energy exceeds the average energy.
+
+However, it relies on only peak values, which may appear by chance and may not reflect the signal’s essential characteristics. Kurtosis, discussed below, also captures the relationship between 'central' and 'marginal' amplitude values, but it considers all values, not just the peaks.
```

Comment:
Also use double quotation marks here for consitency: "central" and "marginal"

Reply:
- done